### PR TITLE
Use relative paths for user display

### PR DESCRIPTION
**Pull request by charliermarsh**
## Summary

This PR changes our user-facing representation for paths to use relative paths, when the path is within the current working directory. This mirrors what we do in Ruff. (If the path is _outside_ the current working directory, we print an absolute path.)

Before:

```shell
❯ uv venv .venv2
Using Python 3.12.2 interpreter at: /Users/crmarsh/workspace/uv/.venv/bin/python3
Creating virtualenv at: .venv2
Activate with: source .venv2/bin/activate
```

After:

```shell
❯ cargo run venv .venv2
    Finished dev [unoptimized + debuginfo] target(s) in 0.15s
     Running `target/debug/uv venv .venv2`
Using Python 3.12.2 interpreter at: .venv/bin/python3
Creating virtualenv at: .venv2
Activate with: source .venv2/bin/activate
```

Note that we still want to use the existing `.simplified_display()` anywhere that the path is being simplified, but _still_ intended for machine consumption (e.g., when passing to `.current_dir()`).

### CODE REVIEW

**Comment by zanieb** [comment]
+1 I prefer this strongly.

I would be careful with cases where the user provides something relative but stepping upwards like `./../../foo`

File: crates/uv-fs/src/path.rs [hunk 1]
```diff
@@ -9,18 +12,19 @@ pub trait Simplified {
 
     /// Render a [`Path`] for user-facing display.
     ///
-    /// On Windows, this will strip the `\\?\` prefix from paths. On other platforms, it's
-    /// equivalent to [`std::path::Display`].
-    fn simplified_display(&self) -> std::path::Display;
+    /// This will relativize the path against the current working directory and, on Windows, strip
+    /// the `\\?\` prefix from paths.
+    fn user_display(&self) -> std::path::Display;
```
Comment (charliermarsh) [inline crates/uv-fs/src/path.rs:17]: I renamed this because it's no longer equivalent to `.simplify().display()`.

File: crates/uv/tests/venv.rs [hunk 2]
```diff
@@ -29,11 +29,8 @@ fn create_venv() -> Result<()> {
             r"Using Python 3\.\d+\.\d+ interpreter at: .+",
             "Using Python [VERSION] interpreter at: [PATH]",
         ),
-        (&filter_venv, "/home/ferris/project/.venv"),
-        (
-            filter_prompt,
-            "Activate with: source /home/ferris/project/.venv/bin/activate",
-        ),
+        (filter_prompt, "Activate with: source .venv/bin/activate"),
+        (&filter_venv, ".venv"),
```
Comment (konstin) [inline crates/uv/tests/venv.rs:33]: Do we still need this filter?

**Review by konstin — APPROVED** [review]
Much better

File: crates/uv/tests/venv.rs (hunk 2 above)
Comment (charliermarsh) [reply crates/uv/tests/venv.rs:33]: Yeah, for some reason on macOS these paths are sometimes prefixed with `/private/`

**Pull request merged**